File: app/src/main/java/android/sample/com/topreposgithub/MainActivity.java
package android.sample.com.topreposgithub;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.LinearLayoutManager;

import android.os.AsyncTask;
import android.os.Bundle;
import android.sample.com.topreposgithub.databinding.MainActivityBinding;

import java.util.List;

public class MainActivity extends AppCompatActivity {

    private MainActivityBinding binding;
    private static final String TOPIC = "android";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.main_activity);

        new RepositoryTask().execute(TOPIC);
    }

    private class RepositoryTask extends AsyncTask<String, Void, List<Repository>> {

        @Override
        protected List<Repository> doInBackground(String... params) {
            RepositoryRemote repositoryRemote = ((TopReposGithubApplication) getApplication()).getRepositoryRemote();
            return repositoryRemote.getTopicRepos(params[0]);
        }

        @Override
        protected void onPostExecute(List<Repository> result) {
            super.onPostExecute(result);
            binding.recycler.setLayoutManager(new LinearLayoutManager(getApplicationContext()));

            RepoAdapter adapter = new RepoAdapter(getApplicationContext(), result);
            binding.recycler.setAdapter(adapter);
        }
    }
}

File: app/src/main/java/android/sample/com/topreposgithub/ui/MainActivity.kt
package android.sample.com.topreposgithub.ui

import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.LinearLayoutManager

import android.os.Bundle
import android.sample.com.topreposgithub.R
import android.sample.com.topreposgithub.databinding.MainActivityBinding
import android.sample.com.topreposgithub.viewmodels.RepoViewModel
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders

class MainActivity: AppCompatActivity() {

    private lateinit var binding:MainActivityBinding
    private var viewModel: RepoViewModel? = null
    private val TOPIC = "topic:android"


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.main_activity)

        viewModel = ViewModelProviders.of(this).get(RepoViewModel::class.java)
        viewModel!!.fetchRepos(TOPIC).observe(this, Observer {
            binding.recycler.layoutManager = LinearLayoutManager(this.applicationContext)
            binding.recycler.adapter = RepoAdapter(it)
        })
    }
}
